packages/destination-actions/src/destinations/fullstory/identifyUserV2/index.ts

```diff
@@ -0,0 +1,71 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import { createUserRequestParams } from '../request-params'
+import { normalizePropertyNames } from '../vars'
+
+const action: ActionDefinition<Settings> = {
+  title: 'Identify User V2',
+  description: 'Sets user identity variables, creates user if it does not exist',
+  platform: 'cloud',
+  defaultSubscription: 'type = "identify"',
+  fields: {
+    uid: {
+      type: 'string',
+      required: true,
+      description: "The user's id",
+      label: 'User ID',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    display_name: {
+      type: 'string',
+      required: false,
+      description: "The user's display name",
+      label: 'Display Name',
+      default: {
+        '@path': '$.traits.name'
+      }
+    },
+    email: {
+      type: 'string',
+      required: false,
+      description: "The user's email",
+      label: 'Email',
+      default: {
+        '@path': '$.traits.email'
+      }
+    },
+    properties: {
+      type: 'object',
+      required: false,
+      description: 'The Segment traits to be forwarded to FullStory',
+      label: 'Traits',
+      default: {
+        '@path': '$.traits'
+      }
+    }
+  },
+  perform: (request, { payload, settings }) => {
+    const { properties, uid, email, display_name } = payload
+
+    const normalizedProperties = normalizePropertyNames(properties, { camelCase: true })
+
+    delete normalizedProperties.email
+    delete normalizedProperties.name
```

Comment:
It feels like we should only delete `normalizedProperties.email` if `traits.email` is mapped to `email` and if `traits.name` is mapped to `display_name`. These are the default mappings, but a user can change these mappings for their destination action.

I don't think we have the ability to inspect mappings in this function. We could ensure `normalizedProperties.email === email` and `normalizedProperties.name === display_name` before deleting. However, that could lead to variable behavior if a customer sometimes provides matching values and other times different values depending on their `identify` payloads and destination mappings. I'm not sure I have a great solution for this.

Reply:
Sounds like it is safer to leave the duplicates.

Reply:
Agreed. Would have been nice to address the duplication, but I'm not sure we can do so in a safe way. Would be good to get Patrick's and Josh's thoughts on leaving the duplication in.

Reply:
Would another solution be to change the API contract such that `display_name` and `email` are reserved fields in the `properties` collection? (and not fields that live outside of properties)?

Reply:
@patrick-fs I'm not sure. Imagine the following scenario:

A customer has applied the default mappings, meaning `traits.name` would be mapped to `display_name`. We can pass `properties.display_name` in the request body with the value given to `traits.name`. However, I don't see a way for us to know that `traits.name` was mapped to `properties.display_name` when the action is executing other than by comparing their values and taking a guess that they probably don't need a `name_str` custom var in FullStory. The fact that `display_name` is nested within `properties` wouldn't seem to help us.

We could change our default mappings such that we'd expect a customer to send `traits.display_name` to Segment instead of `traits.name`. However:
- Customers can change their mappings such that we could end up back in the first scenario where a customer could map `traits.name` to `properties.display_name`. We could potentially argue that the duplication is more warranted in that case because they've diverged from the default mappings.
- `traits.name` may have been purposefully chosen based on common Segment usage patterns. I don't have background here.

Reply:
All right, I got it. So do I have this right:

* The two `delete` statements safe as long as default mappings are preserved
  * `delete normalizedProperties.email`
  * `delete normalizedProperties.email`
* But default mappings can be changed in the Segment UI and we have no way of knowing whether they've changed or what their state is at action execution time
* Therefor, we need to maintain duplicate data (`$. email`, `$.properties.email` + `$.display_name`, `$.properties.name`) to prevent losing data all together

Do I have that right?

Reply:
Yes, you have that right.